client/src/components/NavBar/AuthBars/LoggedInBar.tsx

```diff
@@ -4,15 +4,77 @@ import Typography from '@material-ui/core/Typography';
 import { Link } from 'react-router-dom';
 import useStyles from './useStyles';
 import AuthMenu from '../../AuthMenu/AuthMenu';
+import Notifications from './Notifications/Notifications';
+import { useState, useEffect } from 'react';
+import getUnread from '../../../helpers/APICalls/getAllUnreadNotifications';
+import readNotification from '../../../helpers/APICalls/readNotification';
+
+interface notification {
+  type: string;
+  title: string;
+  description: string;
+  read: boolean;
+  date: Date;
+  image: string;
+  _id: string;
+}
 
 const LoggedInBar = (): JSX.Element => {
   const classes = useStyles();
+  const [unread, setUnread] = useState(false);
+  const [notificationsOpen, setNotificationsOpen] = useState(false);
+  const [notifications, setNotifications] = useState<notification[]>([
+    {
+      type: '',
+      title: 'filler',
+      description: '',
+      read: false,
+      date: new Date('1/1/2000'),
+      image: '',
+      _id: '',
+    },
+  ]);
+
+  useEffect(() => {
+    getUnread().then((data) => {
+      const notificationsData = data.success?.notifications;
+      const notificationsArray: notification[] = [
+        {
+          type: '',
+          title: '',
+          description: '',
+          read: false,
+          date: new Date('1/1/2000'),
+          image: '',
+          _id: '',
+        },
+      ];
+      notificationsData?.forEach((n) => notificationsArray.push(n));
```

Comment:
You don't need to loop over the notifications here, if the API call is successful you can directly set the notifications as the state 👍🏽

Reply:
I tried using the following code inside the useEffect but could not get rid of an error.

      const notificationsData: notification[] = data.success?.notifications;
      setNotifications(notificationsData);
      if (notificationsData.length > 0) {
        setUnread(true);
      }
      

Type '[{ type: string; title: string; description: string; read: boolean; image: string; _id: string; createdAt: Date; }] | undefined' is not assignable to type 'notification[]'. Type 'undefined' is not assignable to type 'notification[]'.

Reply:
The reason you were getting an error is because `data.success?.notifications;` may possibly be undefined. In order solve the problem you can add an 'if' check and then `setNotifications`

```
if (notificationsData) {
   setNotifications(notificationsData);
}
```